vars/runTest.groovy

```diff
@@ -117,33 +117,33 @@ def call(Map config = [:]) {
     // https://issues.jenkins-ci.org/browse/JENKINS-39203
     // Once that is fixed all of the below should be pushed up into the
     // Jenkinsfile post { stable/unstable/failure/etc. }
-    def status = "SUCCESS"
+    String status = 'SUCCESS'
     if (rc != 0) {
-        status = "FAILURE"
+        status = 'FAILURE'
     } else if (rc == 0) {
-        def test_failure = false
-        def test_error = false
+        boolean test_failure = false
+        boolean test_error = false
         if (config['junit_files']) {
-            def filesList = []
+            List filesList = []
             config['junit_files'].split().each {
                 filesList.addAll(findFiles(glob: it))
             }
             if (filesList) {
                 String junit_xml = filesList.collect{"'" + it + "'"}.join(' ')
                 if (sh(label: 'Check junit xml files for errors',
-                       script: 'grep "<error " ' + junit_xml,
+                       script: 'grep -E "<error( |Details>|StackTrace>)" ' + junit_xml,
```

Comment:
Just to be pedantic, this could be further factored down to:

```suggestion
                       script: 'grep -E "<error( |(Details|StackTrace)>)" ' + junit_xml,
```

Reply:
I am sticking with the shorter line.